lib/streamlit/elements/json.py

```diff
@@ -91,23 +106,21 @@ def json(
             body = body.to_dict()
 
         if not isinstance(body, str):
-            # Check if body is iterable, if it is convert it's sets to lists
-            try:
-                iter(body)
-            except TypeError:
-                # body is not iterable, do nothing with the body
-                pass
-            else:
+            # Check if body is iterable, if it is, convert its sets to lists
+            if is_iterable(body):
                 # body is iterable, look for sets and change them to lists
-                body = _convert_sets_to_lists(body)
+                # TODO(harahu): This function does not handle iterables in
+                #  general. Either generalize the function, or do further type
+                #  checking here.
+                body = _convert_sets_to_lists(body)  # type: ignore[arg-type]
```

Comment:
I don't consider it part of the scope of this PR to address this issue beyond making it visible in the code. Hoping someone else wants to pick up the ball from here, after merge.

Reply:
This is a good call out, please see my comment below 👍